app.rb

```diff
@@ -0,0 +1,143 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+  end
+
+  def run
+    actions.each { |key, value| puts "#{key}: #{value}" }
+    action = gets.chomp.to_i
+    action(action)
+  end
+
+  private
+
+  def actions
+    puts # Add a blank line before the actions list
+    {
+      1 => 'List all books',
+      2 => 'List all people',
+      3 => 'Create a person',
+      4 => 'Create a book',
+      5 => 'Create a rental',
+      6 => 'List all rentals for a given person id',
+      7 => 'Exit'
+    }
+  end
+
+  def action(action)
+    action_list = { 1 => 'list_books', 2 => 'list_people', 3 => 'add_person', 4 => 'add_book', 5 => 'add_rental',
+                    6 => 'list_rentals_by_person_id', 7 => 'exit' }
+
+    if action_list.key?(action)
+      send(action_list[action])
+    else
+      puts 'Invalid input!'
+    end
+
+    run # List actions once the user is done with the current action
+  end
+
+  def add_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    type = gets.chomp.to_i
+
+    puts 'Invalide input' if type != 1 && type != 2
+    add_student if type == 1
+    add_teacher if type == 2
+  end
+
+  def list_people(show_index: false)
+    @persons.each_with_index do |person, index|
+      result = "[#{person.class.name}] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+      result = "#{index}) " + result if show_index
+      puts result
+    end
+  end
+
+  def add_student
+    print 'Enter the name of the student: '
+    name = gets.chomp.to_s
+    print 'Enter the age of the student: '
+    age = gets.chomp.to_i
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.to_s.downcase == 'y'
+    student = Student.new(age, name, parent_permission: parent_permission)
+    @persons << student
+
+    puts "Student with id #{student.id} successfully created"
+  end
+
+  def add_teacher
+    print 'Enter the name of the teacher: '
+    name = gets.chomp.to_s
+    print 'Enter the age of the teacher: '
+    age = gets.chomp.to_i
+    print "And teacher's specialization: "
+    specilization = gets.chomp.to_s
+    teacher = Teacher.new(specilization, age, name)
+    @persons << teacher
+
+    puts "Teacher with id #{teacher.id} successfully created"
+  end
+
+  def list_books(show_index: false)
+    @books.each_with_index do |book, index|
+      result = "Title: \"#{book.title}\", Author: #{book.author}"
+      result = " #{index}) " + result if show_index
+      puts result
+    end
+  end
+
+  def add_book
+    print 'Enter the name of the book: '
+    name = gets.chomp.to_s
+    print 'Enter the author of the book: '
+    author = gets.chomp.to_s
+    book = Book.new(name, author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  # rubocop:disable Style/Next
+
+  def list_rentals_by_person_id
```

Comment:
the list_rentals_by_person_id method is not working. Check your code and fix the error.